chatterbot/adapters/logic/multi_adapter.py

```diff
@@ -24,9 +48,11 @@ def process(self, statement):
         result = None
         max_confidence = -1
 
-        for adapter in self.adapters:
-            if adapter.can_process(statement):
-                confidence, output = adapter.process(statement)
+        usable_adapters = self.get_usable_adapters(statement)
+
+        # with ProcessPoolExecutor(max_executors) as executor:
+        with ThreadPoolExecutor(max_workers=len(usable_adapters)) as executor:
+            for confidence, output, adapter in executor.map(self.execute_adapter_process, usable_adapters):
```

Comment:
The executor's `map` function does execute the function asynchronously with each of the values in the iterable. (https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor.map). From what I understand, the function execution via `map` is asynchronous, but the for loop is still synchronous.